webrtc/RTCDtlsTransport-getRemoteCertificates.html

```diff
@@ -4,104 +4,97 @@
 <script src="/resources/testharness.js"></script>
 <script src="/resources/testharnessreport.js"></script>
 <script src="RTCPeerConnection-helper.js"></script>
+<script src="RTCTransport-helper.js"></script>
 <script>
   'use strict';
 
   // The following helper functions are called from RTCPeerConnection-helper.js:
-  //   exchangeIceCandidates
-  //   doSignalingHandshake
-
-  /*
-    5.5.  RTCDtlsTransport Interface
-      interface RTCDtlsTransport : EventTarget {
-        readonly attribute RTCDtlsTransportState state;
-        sequence<ArrayBuffer> getRemoteCertificates();
-                 attribute EventHandler          onstatechange;
-                 attribute EventHandler          onerror;
-        ...
-      };
-
-      enum RTCDtlsTransportState {
-        "new",
-        "connecting",
-        "connected",
-        "closed",
-        "failed"
-      };
-
-      getRemoteCertificates
-        Returns the certificate chain in use by the remote side, with each certificate
-        encoded in binary Distinguished Encoding Rules (DER) [X690].
-        getRemoteCertificates() will return an empty list prior to selection of the
-        remote certificate, which will be completed by the time RTCDtlsTransportState
-        transitions to "connected".
-   */
-  async_test(t => {
+  //  exchangeIceCandidates
+  //  doSignalingHandshake
+
+  // The following helper functions are called from RTCTransport-helper.js:
+  //  waitForConnectedState
+  //  getDtlsTransportFromSctpTransport
+  //  getDtlsTransportsFromSenderReceiver
+
+  function validateConnectedDtlsTransport(dtlsTransport) {
+    const certs = dtlsTransport.getRemoteCertificates();
+
+    assert_greater_than(certs.length, 0,
+      'Expect DTLS transport to have at least one remote certificate when connected');
+
+    for(const cert of certs) {
+      assert_true(cert instanceof ArrayBuffer,
+        'Expect certificate elements be instance of ArrayBuffer');
+    }
+  }
+
+  function validateConnectingDtlsTransport(dtlsTransport) {
+    if (dtlsTransport.state !== 'connected') {
```

Comment:
This will still succeed if the state is already connected when you get here - so you're not necessarily checking what the test looks like it's checking.
Better: Assert that the state is not "connected"

Reply:
We can't make sure that the dtlsTransport does not become connected when it reach this part of the code. There is a race condition of the transport becoming connected in parallel with the async tests. Since the tests are running locally, there is a chance that the whole connection is connected before the code resume from `await doSignalingHandshake(...)`.

If we want the test to run consistently, the alternative is to remove `validateConnectingDtlsTransport()` altogether.

Reply:
state changes will happen only while you await something. the only place where you await stuff before this is in doSignalingHandshake. But it's possible that the connection might connect while you're awaiting doSignalingHandshake. OK, let's do it this way.